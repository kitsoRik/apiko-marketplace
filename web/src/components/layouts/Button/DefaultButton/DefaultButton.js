import React from 'react';

import "./DefaultButton.scss";

const DefaultButton = ({ className, style, value, disabled, ...props }) => {
    return (
        <button
            className={`button-default ${className ?? ""}`}
            style={style}
            disabled={disabled ? true : null}
            {...props}
        >
            {value}
        </button>
    )
};

export default DefaultButton;