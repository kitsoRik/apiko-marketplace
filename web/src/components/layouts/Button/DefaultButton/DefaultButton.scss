.button-default {
    padding: 20px 50px;

    border: none;
    color: #fff;
    background-color: #349a89;

    border-radius: 5px;

    font-size: 14px;

    &:hover {
        background-color: red;
    }

    &:active {
        background-color: #f2b;
    }

    &[disabled] {
        background-color: gray;
    }
}
