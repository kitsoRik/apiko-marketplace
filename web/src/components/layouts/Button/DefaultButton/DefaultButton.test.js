import React from "react";
import { create } from "react-test-renderer";
import DefaultButton from "./DefaultButton";
import { mount } from "enzyme";

describe("Default button", () => {
	it("should render without value", () => {
		const component = create(<DefaultButton />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with value", () => {
		const component = create(<DefaultButton value="My custom value" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const component = create(
			<DefaultButton value={"My custom calue"} disabled={true} />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should have class", () => {
		const component = mount(<DefaultButton />);
		expect(component.find("button").hasClass("button-default")).toEqual(
			true
		);
	});

	it("should skip click event", () => {
		const onClick = jest.fn();
		const component = mount(
			<DefaultButton onClick={onClick} disabled={true} />
		);
		component.find("button").simulate("click");
		expect(onClick).toBeCalledTimes(0);
	});

	it("should emit click event", () => {
		const onClick = jest.fn();
		const component = mount(<DefaultButton onClick={onClick} />);
		const button = component.find("button");

		button.simulate("click");
		expect(onClick).toBeCalledTimes(1);
		button.simulate("click");
		expect(onClick).toBeCalledTimes(2);
	});
});
