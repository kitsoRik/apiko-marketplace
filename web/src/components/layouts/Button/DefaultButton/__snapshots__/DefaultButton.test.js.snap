// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default button should render disabled 1`] = `
<button
  className="button-default "
  disabled={true}
>
  My custom calue
</button>
`;

exports[`Default button should render with value 1`] = `
<button
  className="button-default "
  disabled={null}
>
  My custom value
</button>
`;

exports[`Default button should render without value 1`] = `
<button
  className="button-default "
  disabled={null}
/>
`;
